### DISCUSSION TITLE
ISTIO_mTLS with Gateway vs Sidecar

### DISCUSSION THREAD

**Discussion by haithamshahin333** [istio/istio#49075] [Q&A]
Hello,

I've enabled a federated mesh using Spire, I'm seeing cluster1 in trust domain foo.com can do ISTIO_MTLS with an ingress gateway win cluster2 in trust domain bar.com. However, when I configure the gateway to PASSTHROUGH such that ISTIO_MTLS should happen between the two services in each cluster, I see a OpenSSL error.

Is there a difference in how the gateway does ISTIO_MTLS vs a sidecar? If you refer to the image below, what I'm suggesting is that the only real way to get this working is to place a gateway between the two services and have ISTIO_MTLS occur at the gateway.

@howardjohn @linsun any help would be appreciated! This is associated with the Spire Federation issues you all have commented on in the past. Thanks in advance.

![image](https://github.com/istio/istio/assets/42976403/9cd57ca8-edfa-4e59-8520-606fbbae91f9)

**Comment by yann-soubeyrand** [comment]
Hi @haithamshahin333, could you clarify how you made services in both clusters communicate with each others?

**Reply by rootsongjc** [reply]
How did you set up the CA for both clusters? @haithamshahin333
